tests/test_hdfs.py

```diff
@@ -34,16 +44,15 @@ def hdfs_with_kerberos():
             "data-2-node",
         ]
         principals = []
-        for instance in instances:
-            for primary in primaries:
-                principals.append(
-                    "{primary}/{instance}.{fqdn}@{REALM}".format(
-                        primary=primary,
-                        instance=instance,
-                        fqdn=fqdn,
-                        REALM=sdk_auth.REALM
-                    )
+        for instance_primary in itertools.product(instances, primaries):
```

Comment:
not a blocker, more of a nit: since we're doing a `product` it'd be semantically cleaner to explicitly split the elements into a tuple instead of relying on tuple indexing

```
for (instance, primary) in ...
```

Reply:
Agree. I'll do that.